framework/src/signal.c

```diff
@@ -0,0 +1,30 @@
+//
+// Created by tywael on 2/12/23.
+//
+
+#include "libunit.h"
+
+int catch_res(int signal)
```

Comment:
Is it a better idea ?
```suggestion
enum e_result catch_res(int signal)
```